src/main/java/org/fogbowcloud/app/jes/arrebol/JobSynchronizer.java

```diff
@@ -1,60 +1,73 @@
 package org.fogbowcloud.app.jes.arrebol;
 
-import com.google.gson.Gson;
-import java.util.List;
-import java.util.Map;
-import java.util.Objects;
-import java.util.Properties;
 import org.apache.log4j.Logger;
 import org.fogbowcloud.app.core.command.Command;
 import org.fogbowcloud.app.core.command.CommandState;
 import org.fogbowcloud.app.core.task.Task;
 import org.fogbowcloud.app.core.task.TaskState;
 import org.fogbowcloud.app.jdfcompiler.job.JDFJob;
 import org.fogbowcloud.app.jdfcompiler.job.JobState;
-import org.fogbowcloud.app.jes.arrebol.models.ArrebolCommand;
-import org.fogbowcloud.app.jes.arrebol.models.ArrebolCommandState;
-import org.fogbowcloud.app.jes.arrebol.models.ArrebolJob;
-import org.fogbowcloud.app.jes.arrebol.models.ArrebolTask;
-import org.fogbowcloud.app.jes.arrebol.models.ArrebolTaskState;
-import org.fogbowcloud.app.jes.arrebol.models.ExecutionState;
-import org.fogbowcloud.app.jes.exceptions.JobStatusException;
+import org.fogbowcloud.app.jes.JobExecutionService;
+import org.fogbowcloud.app.jes.arrebol.models.*;
+import org.fogbowcloud.app.jes.exceptions.ArrebolConnectException;
+import org.fogbowcloud.app.jes.exceptions.JobExecStatusException;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static java.lang.Thread.sleep;
 
-public class ArrebolSynchronizer implements Synchronizer<JDFJob> {
+/** Sync local job state with it execution. */
+public class JobSynchronizer implements Synchronizer<JDFJob> {
 
-    private static final Logger logger = Logger.getLogger(ArrebolSynchronizer.class);
+    private static final Logger logger = Logger.getLogger(JobSynchronizer.class);
 
-    private final ArrebolRequestsHelper requestsHelper;
+    private final JobExecutionService jobExecutionService;
 
-    public ArrebolSynchronizer(Properties properties) {
-        this.requestsHelper = new ArrebolRequestsHelper(properties);
+    public JobSynchronizer(JobExecutionService jobExecutionService) {
+        this.jobExecutionService = jobExecutionService;
     }
 
     @Override
     public JDFJob sync(JDFJob job) {
 
         final String executionId = job.getExecutionId();
         if (Objects.nonNull(executionId) && !executionId.trim().isEmpty()) {
-            try {
-                String jobExecutionJson = this.requestsHelper.statusArrebolJob(executionId);
-                logger.debug("JSON Response [" + jobExecutionJson + "]");
-                Gson gson = new Gson();
-                ArrebolJob arrebolJob = gson.fromJson(jobExecutionJson, ArrebolJob.class);
-                this.updateJob(job, arrebolJob);
-            } catch (Exception e) {
-                throw new JobStatusException(e.getMessage());
+            JobExecArrebol jobExecArrebol = null;
+            while (Objects.isNull(jobExecArrebol)) {
```

Comment:
Why needed this while?

Reply:
To retry the status operation of a job and only stop trying when there is a response from Arrebol.

Reply:
Solved in 89f1f1d7ac916357cd8819e7f6200c1eb40fb875.